Kernel/Arch/riscv64/CSR.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2023, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/BitCast.h>
+#include <AK/Types.h>
+
+#include <AK/Platform.h>
+VALIDATE_IS_RISCV64()
+
+// Documentation for the CSRs:
+// RISC-V ISA Manual, Volume II (https://github.com/riscv/riscv-isa-manual/releases/download/Priv-v1.12/riscv-privileged-20211203.pdf)
+
+namespace Kernel::RISCV64::CSR {
+
+// 2.2 CSR Listing
+enum class Address : u16 {
+    // Supervisor Trap Setup
+    SSTATUS = 0x100,
+    SIE = 0x104,
+    STVEC = 0x105,
+
+    // Supervisor Protection and Translation
+    SATP = 0x180,
+
+    // Unprivileged Counters/Timers
+    TIME = 0xc01,
+};
+
+inline FlatPtr read(Address address)
+{
+    FlatPtr ret;
+    asm volatile("csrr %0, %1"
+                 : "=r"(ret)
+                 : "i"(address));
+    return ret;
+}
```

Comment:
Is it necessary for normal operation to write these CSRs after initalization? For x86_64, we intentionally put operations like writing to cr4 into non-inlined `UNMAP_AFTER_INIT` functions in order to reduce the number of spicy ROP gadgets laying around in the kernel.